src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -682,18 +747,23 @@ ErrorIfInsertSelectQueryNotSupported(Query *queryTree, RangeTblEntry *insertRte,
 	{
 		if (!allReferenceTables)
 		{
-			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-							errmsg("If data inserted into a reference table, "
-								   "all of the participating tables in the "
-								   "INSERT INTO ... SELECT query should be "
-								   "reference tables.")));
+			return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
+								 "may only reference other reference tables when "
+								 "targeting reference tables with INSERT ... SELECT",
```

Comment:
Well, the former violates the message guidelines (full sentence, upper case start) and doesn't actually contain more information?